src/protvista-uniprot.ts

```diff
@@ -209,68 +217,102 @@ class ProtvistaUniprot extends LitElement {
     loadComponent('protvista-uniprot-structure', ProtvistaUniprotStructure);
   }
 
-  _loadData() {
+  async _loadData() {
     const accession = this.accession;
     if (accession && this.config) {
-      const tasks = this.config.categories.map(
-        ({ name, url, adapter, tracks }) => {
-          const urlWithProtein =
-            url.indexOf('{}') >= 0
-              ? url.replace('{}', accession)
-              : `${url}${accession}`;
-          // TODO: remove this conditional setTimeout when InterPro API more stable
-          // NOTE: this is just to ensure the InterPro fetches are enqueued last
-          return sleep(/interpro/.test(urlWithProtein) ? 100 : 0)
-            .then(() => load(urlWithProtein))
-            .then(({ payload }) => {
-              if (!payload) return;
-              const data = adapter ? adapters[adapter](payload) : payload;
-              this.data[name] =
-                adapter === 'protvista-feature-adapter'
-                  ? data.filter(
-                      ({ category }: { category?: string }) =>
-                        !category || category === name
+      // Get the list of unique urls
+      const urls = this.config.categories
+        .map(({ tracks }) => tracks.map(({ data }) => data[0].url))
+        .flat();
+      const uniqueUrls = [...new Set(urls)];
+      // Get the data for all urls and store it
+      for (const url of uniqueUrls) {
+        try {
+          const data = await load(url.replace('{accession}', accession));
+          this.rawData[url] = data.payload;
+        } catch (e) {
+          // TODO handle this better based on error code
+          // Fail silently for now
+        }
+      }
```

Comment:
Not sure about waiting for the previous data before storing it and then fetching the second data, and so on and so on. We could probably fetch everything at once and let user browser parallelise as it sees fit.
I would probably replace with something like this:
```suggestion
      await Promise.all(
        uniqueUrls.map((url) =>
          load(url.replace('{accession}', accession)).then(
            (data) => (this.rawData[url] = data.payload),
            // TODO handle this better based on error code
            // Fail silently for now
            (error) => console.warn(error)
          )
        )
      );
```
Make sure one error is not blocking the function to continue. If it does, the whole block of code probably needs a try/catch

Reply:
Yep, definitely running the requests in parallel 👍 
I've tried it with examples with no data and it seems to be ok without the try/catch